apps/spruce/src/gql/client/useCreateGQLClient/index.ts

```diff
@@ -17,15 +17,20 @@ import {
 import { secretFieldsReq } from "gql/fetch";
 import { SecretFieldsQuery } from "gql/generated/types";
 import { environmentVariables } from "utils";
+import { getCorpLoginURL, isRemoteEnv } from "utils/environmentVariables";
 import { leaveBreadcrumb, SentryBreadcrumb } from "utils/errorReporting";
 
 const { getGQLUrl } = environmentVariables;
 
-export const useCreateGQLClient = (): ApolloClient<NormalizedCacheObject> => {
+export const useCreateGQLClient = ():
+  | ApolloClient<NormalizedCacheObject>
+  | undefined => {
```

Comment:
when does it return `undefined`?

Reply:
It is initially undefined until after the secretFields query responds. https://github.com/khelif96/ui/blob/76880083df11136f85ddcaa2305241fb3f16179b/apps/spruce/src/gql/client/useCreateGQLClient/index.ts#L31